scripts/preconversion_assessment_script.py

```diff
@@ -154,34 +156,93 @@ def setup_convert2rhel(required_files=None):
                 os.chmod(required_file.path, 0o644)
 
 
+# Code taken from
+# https://github.com/oamg/convert2rhel/blob/v1.4.1/convert2rhel/utils.py#L345
+# and modified to adapt the needs of the tools that are being executed in this
+# script.
+def run_subprocess(cmd, print_cmd=True, env=None):
+    """
+    Call the passed command and optionally log the called command
+    (print_cmd=True) and environment variables in form of dictionary(env=None).
+    Switching off printing the command can be useful in case it contains a
+    password in plain text.
+
+    The cmd is specified as a list starting with the command and followed by a
+    list of arguments. Example: ["yum", "install", "<package>"]
+    """
+    # This check is here because we passed in strings in the past and changed
+    # to a list for security hardening.  Remove this once everyone is
+    # comfortable with using a list instead.
+    if isinstance(cmd, str):
+        raise TypeError("cmd should be a list, not a str")
+
+    if print_cmd:
+        print("Calling command '%s'" % " ".join(cmd))
+
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, bufsize=1, env=env
+    )
+    output = ""
+    for line in iter(process.stdout.readline, b""):
+        line = line.decode("utf8")
+        output += line
+
+    # Call wait() to wait for the process to terminate so that we can
+    # get the return code.
+    process.wait()
+
+    return output, process.returncode
+
+
 def install_convert2rhel():
-    subprocess.check_output(["yum", "install", "convert2rhel", "-y"])
-    subprocess.check_output(["yum", "update", "convert2rhel", "-y"])
+    """Install the convert2rhel tool to the system."""
+    output, returncode = run_subprocess(
+        ["yum", "install", "convert2rhel", "-y"],
+    )
+    if returncode:
+        print(
```

Comment:
Yeah, because that will be collected by the stdout of the worker. So, printing to the stdout is okay for now. 

We can take a look at a logging solution later to make it more robust, but for now, a simple print statement would be enough.